t/60-cidrvalidate.t has started to fail (Net-Whois-IANA-0.47)

As per subject. Looks like Whois DB has changed. Sample fail report: http://www.cpantesters.org/cpan/report/4339c13c-9ab1-11ec-8208-44482222c06c

Thanks for reporting this issue... 

I'm not sure if we should catch up and keep that test in `integration` mode or freeze it for stability lol
going to have a look at it, I would probably want to avoid fixing every once in a while lol

```
t/01-connect-query.t ... ok
t/10-ripe.t ............ ok
t/20-arin.t ............ ok
t/30-apnic.t ........... ok
t/40-lacnic.t .......... ok
t/50-afrinic.t ......... ok

# Failed test 'cidr'
# at t/60-cidrvalidate.t line 29.
# +---------------+----+----------------------------+
# | GOT           | OP | CHECK                      |
# +---------------+----+----------------------------+
# | 191.96.0.0/16 | =~ | (?^:^191\.96\.58(\.0)?/24) |
# +---------------+----+----------------------------+
# Seeded srand with seed '20220303' from local date.
t/60-cidrvalidate.t .... 
Dubious, test returned 1 (wstat 256, 0x100)
Failed 1/4 subtests 
t/90-misc.t ............ ok
t/accessors.t .......... ok
t/boilerplate.t ........ skipped: do not run boilerplate for AUTOMATED_TESTING
t/manifest.t ........... skipped: Author tests not required for installation
t/pod.t ................ ok

Test Summary Report
-------------------
t/60-cidrvalidate.t  (Wstat: 256 Tests: 4 Failed: 1)
  Failed test:  3
  Non-zero exit status: 1
Files=14, Tests=59,  9 wallclock secs ( 0.03 usr  0.05 sys +  0.98 cusr  0.34 csys =  1.40 CPU)
Result: FAIL
Failed 1/14 test programs. 1/59 subtests failed.
make: *** [Makefile:880: test_dynamic] Error 255
```

--- a/Changes
+++ b/Changes
@@ -1,5 +1,7 @@
 {{$NEXT}}
 
+    - Fix t/60-cidrvalidate.t
+
 0.47      2021-10-04 10:33:55-06:00 America/Denver
 
     - Reformat POD to avoid verbatim blocks

--- a/Makefile.PL
+++ b/Makefile.PL
@@ -36,7 +36,7 @@ my %WriteMakefileArgs = (
     "Test::MockModule" => "0.175",
     "Test::More" => 0
   },
-  "VERSION" => "0.47",
+  "VERSION" => "0.48",
   "test" => {
     "TESTS" => "t/*.t"
   }

--- a/README.md
+++ b/README.md
@@ -4,7 +4,7 @@ Net::Whois::IANA - Net::Whois::IANA - A universal WHOIS data extractor.
 
 # VERSION
 
-version 0.47
+version 0.48
 
 # SYNOPSIS
 

--- a/t/60-cidrvalidate.t
+++ b/t/60-cidrvalidate.t
@@ -10,6 +10,27 @@ use Test2::Plugin::NoWarnings;
 use Net::CIDR;
 use Net::Whois::IANA;
 
+use Test::MockModule;
+
+# note: we are mocking some internal to freeze the ripe reply
+my $mocked_iana = Test::MockModule->new('Net::Whois::IANA');
+
+$mocked_iana->redefine(
+    source_connect => sub {
+        my ( $self, $source_name ) = @_;
+
+        note "mocked source_connect for ", $source_name;
+        my $sock = $mocked_iana->original('source_connect')->( $self, $source_name );
+
+        $self->{query_sub} = sub {
+            return %{ _ripe_query() } if $source_name eq 'ripe';
+            return ();
+        };
+
+        return $sock;
+    }
+);
+
 my $iana = Net::Whois::IANA->new;
 
 ok $iana;
@@ -26,7 +47,7 @@ if ($cidr) {    # some smokers are flapping... this is kind of a big todo...
 
     if ( defined $cidr->[0] ) {    # query can fail on some smokers...
 
-        like $cidr->[0], qr{^191\.96\.58(\.0)?/24}, 'cidr';
+        like $cidr->[0], qr{^\Q191.96.0.0/16\E}, 'cidr';
 
         # view on travis smoker this can be either 191.96.58/24 or 191.96.58-191.96.58...
         ### need to investigate
@@ -38,3 +59,60 @@ if ($cidr) {    # some smokers are flapping... this is kind of a big todo...
 }
 
 done_testing;
+exit;
+
+sub _ripe_query {
+    return {
+        'abuse-c'  => 'IPXO834',
+        'admin-c'  => 'CYB777',
+        'cidr'     => ['191.96.0.0/16'],
+        'country'  => 'AE',
+        'created'  => '2022-03-02T14:23:29Z',
+        'fullinfo' => '% This is the RIPE Database query service.
+    % The objects are in RPSL format.
+    %
+    % The RIPE Database is subject to Terms and Conditions.
+    % See http://www.ripe.net/db/support/db-terms-conditions.pdf
+
+    % Note: this output has been filtered.
+    %       To receive output for a database update, use the "-B" flag.
+
+    % Information related to \'191.96.0.0 - 191.96.255.255\'
+
+    % Abuse contact for \'191.96.0.0 - 191.96.255.255\' is \'abuse@ipxo.com\'
+
+    inetnum:        191.96.0.0 - 191.96.255.255
+    netname:        AE-CYBERASSETS-20131024
+    country:        AE
+    org:            ORG-CAF4-RIPE
+    admin-c:        CYB777
+    tech-c:         CYB777
+    abuse-c:        IPXO834
+    mnt-lower:      IPXO-MNT
+    mnt-domains:    IPXO-MNT
+    mnt-routes:     IPXO-MNT
+    status:         ALLOCATED PA
+    mnt-by:         CYBER-A-MNT
+    mnt-by:         RIPE-NCC-HM-MNT
+    created:        2022-03-02T14:23:29Z
+    last-modified:  2022-03-02T14:29:23Z
+    source:         RIPE
+
+    % This query was served by the RIPE Database Query Service version 1.102.2 (ANGUS)
+
+
+    ',
+        'inetnum'       => '191.96.0.0 - 191.96.255.255',
+        'last-modified' => '2022-03-02T14:29:23Z',
+        'mnt-by'        => 'CYBER-A-MNT RIPE-NCC-HM-MNT',
+        'mnt-domains'   => 'IPXO-MNT',
+        'mnt-lower'     => 'IPXO-MNT',
+        'mnt-routes'    => 'IPXO-MNT',
+        'netname'       => 'AE-CYBERASSETS-20131024',
+        'org'           => 'ORG-CAF4-RIPE',
+        'permission'    => 'allowed',
+        'source'        => 'RIPE',
+        'status'        => 'ALLOCATED PA',
+        'tech-c'        => 'CYB777'
+    };
+}

